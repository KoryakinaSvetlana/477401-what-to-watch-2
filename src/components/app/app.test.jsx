import React from "react";
import renderer from "react-test-renderer";
import {Route} from 'react-router';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {BrowserRouter as Router} from "react-router-dom";

import App from "./app.jsx";
import NotFound from "../not-found/not-found.jsx";

Enzyme.configure({adapter: new Adapter()});

it(`App correctly renders`, () => {
  const tree = renderer.create(
      <App
        films={[{id: 1, name: `1`}, {id: 2, name: `2`}, {id: 3, name: `3`}]}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});


it(`invalid path should redirect to NotFound page`, () => {
  const wrapper = shallow(<Router><App
    films={[{id: 1, name: `1`}, {id: 2, name: `2`}, {id: 3, name: `3`}]}
  /></Router>);
  const pathMapTest = wrapper.find(Route).reduce((pathMap, route) => {
    const routeProps = route.props();
    pathMap[routeProps.path] = routeProps.component;
    return pathMap;
  }, {});

  expect(pathMapTest[`/somepath`]).toBe(NotFound);
});

