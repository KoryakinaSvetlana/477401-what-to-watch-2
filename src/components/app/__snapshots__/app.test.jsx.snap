// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="1"
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        href="/films/1"
        onClick={[Function]}
      >
        1
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="2"
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        href="/films/2"
        onClick={[Function]}
      >
        2
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="3"
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        href="/films/3"
        onClick={[Function]}
      >
        3
      </a>
    </h3>
  </article>
</div>
`;
